### PR TITLE
🛠️ Ignition Utils

### PR DESCRIPTION
**Pull request by chapulina**
* Generated jobs with: https://build.osrfoundation.org/job/_dsl_ignition/1082/
* Ubuntu CI is working: https://build.osrfoundation.org/job/ignition_utils-ci-pr_any-ubuntu_auto-amd64/1/console
* Mac CI has some Homebrew issues: https://build.osrfoundation.org/job/ignition_utils-ci-pr_any-homebrew-amd64/2/console
* Windows CI has issues that I think need to be fixed on PR https://github.com/ignitionrobotics/ign-utils/pull/1: https://build.osrfoundation.org/job/ignition_utils-ci-pr_any-windows7-amd64/3/console

I'm not sure how to setup the GitHub statuses @j-rivero . How do those work?

### CODE REVIEW

**Comment by j-rivero** [comment]
> I'm not sure how to setup the GitHub statuses @j-rivero . How do those work?

For security reasons we have disabled the auto-connection from github new repositories to our Jenkins. The step is really easy anyway: open any ignition repository configuration for Webhooks that is already working and apply exactly the same settings to the new repo. (just copied from ign-math) https://github.com/ignitionrobotics/ign-util/settings/hooks

File: jenkins-scripts/docker/ign_utils-compilation.bash [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  exit 1
+fi
+
+if [[ -z ${DISTRO} ]]; then
+  echo "DISTRO variable not set!"
+  exit 1
+fi
+
+export BUILDING_SOFTWARE_DIRECTORY="ign-utils"
+export BUILDING_PKG_DEPENDENCIES_VAR_NAME="IGN_UTILS_DEPENDENCIES"
```
Comment (j-rivero) [inline jenkins-scripts/docker/ign_utils-compilation.bash:18]: There is definition of `IGN_UTILS_DEPENDENCIES`. Should go in the dependencies_archive together with the other deps.

File: jenkins-scripts/docker/ign_utils-compilation.bash [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  python3 ${SCRIPT_DIR}/../tools/detect_cmake_major_version.py \
+  ${WORKSPACE}/ign-utils/CMakeLists.txt)
+
+# Check IGN_UTILS version is integer
+if ! [[ ${IGN_UTILS_MAJOR_VERSION} =~ ^-?[0-9]+$ ]]; then
+  echo "Error! IGN_UTILS_MAJOR_VERSION is not an integer, check the detection"
+  exit -1
+fi
+
+if [[ ${IGN_UTILS_MAJOR_VERSION} -ge 6 ]]; then
```
Comment (j-rivero) [inline jenkins-scripts/docker/ign_utils-compilation.bash:31]: Not sure if ign-utils needs c++17 but I'm pretty sure we don't have a 6 major version of it.

**Review by j-rivero — COMMENTED** [review]
A couple of minor things, looks good. Testing builds:

 * Windows: [Build Status](https://build.osrfoundation.org/job/ign_utils-pr-win/2/)
 * Linux: [Build Status](https://build.osrfoundation.org/job/ignition_utils-ci-pr_any-windows7-amd64/8/)

File: jenkins-scripts/docker/ign_utils-compilation.bash (hunk 2 above)
Comment (chapulina) [reply jenkins-scripts/docker/ign_utils-compilation.bash]: 271118f

File: jenkins-scripts/docker/ign_utils-compilation.bash (hunk 1 above)
Comment (chapulina) [reply jenkins-scripts/docker/ign_utils-compilation.bash:18]: 271118f

**Review by j-rivero — APPROVED** [review]

**Pull request merged**